### PR TITLE
feat(tasks): pass downcasted error from panicked task

### LINKED ISSUE #3209: Downcast panic error as string and attach to error message
### Describe the feature

atm we only emit the task name:

https://github.com/paradigmxyz/reth/blob/88b076dbdb470a6c75548f91beba18daff130c7b/crates/tasks/src/lib.rs#L302-L303

the trace will only log `Any` because the res is `Box<Any>`

we should also include the downcasted error message in the channel

ref https://stackoverflow.com/questions/68556819/transform-backtrace-to-string-during-catch-unwind

## TODO
* downcast error message
* send as part of the channel message, requires a message name `PanicMessage {task_name: &str, error: Option<String>}`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shekhirin**
Resolves https://github.com/paradigmxyz/reth/issues/3209

### CODE REVIEW

File: crates/tasks/src/lib.rs [hunk 1]
```diff
@@ -177,14 +179,41 @@ impl Future for TaskManager {
...
+    error: Option<String>,
+}
+
+impl Display for PanickedTaskError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let task_name = self.task_name;
+        if let Some(error) = &self.error {
+            write!(f, "Critical task `{task_name}` panicked: `{error}`")
+        } else {
+            write!(f, "Critical task `{task_name}` panicked")
```
Comment (mattsse) [inline crates/tasks/src/lib.rs:199]: nit:
use shared `write!(f, "Critical task `{task_name}` panicked")` always and `write!(f, ": `{error}`")` in if let

File: crates/tasks/src/lib.rs [hunk 2]
```diff
@@ -428,7 +459,8 @@ mod tests {
 
         runtime.block_on(async move {
             let err = manager.await;
-            assert_eq!(err.0, "this is a critical task");
+            assert_eq!(err.task_name, "this is a critical task");
+            assert_eq!(err.error, Some("intentionally panic".to_string()));
```
Comment (mattsse) [inline crates/tasks/src/lib.rs:463]: cool this actually works

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit.

I wonder if it's also possible to catch the location via backtrace, but unfamiliar with this.

File: crates/tasks/src/lib.rs (hunk 1 above)
Comment (shekhirin) [reply crates/tasks/src/lib.rs:199]: yeah, I wrote it like that at first but didn't like the dangling `:` in the second write

Comment (mattsse) [reply crates/tasks/src/lib.rs:199]: I guess this doesn't make much difference

**Review by mattsse — APPROVED** [review]
writ! nit optional

**Pull request merged**